### PR TITLE
[Android] Navigation icon blink fix - shell (#18867)

### LINKED ISSUE #18867: [.NET 8]Shell's return arrow appears on the mainpage.
### Description

When using Shell navigation to return to the main page from other pages, a return arrow will briefly appear on the left side of the Shell on the main interface. This problem is currently only found on the Android platform (the framework is .NET 8) in .NET 7 and .NET 6 There is no such problem above, here is a video example:

https://github.com/dotnet/maui/assets/63104968/b11c601d-3b70-423a-82e4-5b0e05f6fbad

### Steps to Reproduce

1.Create a .NET MAUI ContentPage(XAML), and keep the default page for the new page.
2.Follow https://learn.microsoft.com/zh-cn/dotnet/maui/fundamentals/shell/navigation?view=net-maui-8.0 to register the route of the newly created ContentPage.Like this:
`Routing.RegisterRoute(nameof(NewPage), typeof(NewPage));`
3.Change the background code of the "Click Me" button to the execution navigation that jumps to the new page. Like this: `Shell.Current.GoToAsync(nameof(NewPage));`
4.Compile and run on Android platform. Just like the uploaded video, click the "Click Me" button to jump to a new page and return to the main interface. A return arrow will briefly appear on the left side of the Shell on the main interface.

### Link to public reproduction project repository

https://github.com/GXSA-Qi/MAUIIssueInShell.git

### Version with bug

8.0.3

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

7.0.101

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #22649: Flyout shown for a moment when interacting with a backbutton in Android Shell
### Description

Visual studio communi5y 2022 17.10.0
.Net Maui 8
Android
Shell

When you go back from Page2 you can see that the Mainpage flyout icon is displayed for a moment.

Flyout is disabled in the AppShell.
 Shell.FlyoutBehavior="Disabled"

This behavior occurs when the Mainpage has a defined title.

1. The bug originates from the back button.

2. When you return to the Mainpage you see the flyout icon, meaning the flyout is deactivated.

3. Here I return to 100% Mainpage view and you can see the flyout icon for a thousandths of seconds it is displayed.

4. And after a few thousandths of a second the flyout icon disappears, which means that the flyout in the shell is disabled.

5. I am attaching a video with 0.5x speed so you can observe the behavior.

https://github.com/dotnet/maui/assets/70722700/876c11ff-169a-4610-ba3c-8a28c7742f57

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

https://github.com/Cr15Denis/BugFlyoutBackbutton

### Version with bug

8.0.40 SR5

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/18867

Fixes https://github.com/dotnet/maui/issues/22649

|Before|After|
|--|--|
|||

### CODE REVIEW

**Comment by AnthonyH7386** [comment]
Any idea when this will be approved?  Such a frustrating bug!

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs [hunk 1]
```diff
@@ -77,6 +77,9 @@ public static void UpdateTitleIcon(this AToolbar nativeToolbar, Toolbar toolbar)
 
 		public static void UpdateBackButton(this AToolbar nativeToolbar, Toolbar toolbar)
 		{
+			//Setting it to null to avoid the icon's blink
+			nativeToolbar.NavigationIcon = null;
```
Comment (PureWeen) [inline src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:81]: Interesting, do we know by chance where this is getting set upstream? I'm curious if we can fix this by not setting this upstream somewhere.

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs (hunk 1 above)
Comment (kubaflo) [reply src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:81]: I tried to look for it but couldn't find a better solution

**Comment by ja3cob** [comment]
Please merge, such an annoying bug @PureWeen

**Comment by AaronBastian** [comment]
@PureWeen What must be true to get this resolved? This is something that is plaguing our production app as well.

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
- failing test unrelated

**Pull request merged**